### PR TITLE
Add env_vars to extra_arguments terraform block

### PR DESCRIPTION
**Pull request by geriBatai**
This PR implements `env_vars` support as described in Issue #319 .

### CODE REVIEW

File: cli/args.go [hunk 1]
```diff
@@ -142,6 +142,18 @@ func filterTerraformExtraArgs(terragruntOptions *options.TerragruntOptions, terr
 	return out
 }
 
+func filterTerraformEnvVars(terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) map[string]string {
```
Comment (brikis98) [inline cli/args.go]: Shouldn't this method only add env vars for specific commands? See `filterTerraformExtraArgs`.

File: cli/args.go [hunk 2]
```diff
@@ -142,6 +142,18 @@ func filterTerraformExtraArgs(terragruntOptions *options.TerragruntOptions, terr
 	return out
 }
 
+func filterTerraformEnvVarsFromExtraArgs(terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) map[string]string {
```
Comment (brikis98) [inline cli/args.go]: This method will apply the env vars to every command. However, if `commands = ["apply"]` is specified, those env vars should only be added to the `apply` command. Could you do the command name check the same way as in `filterTerraformExtraArgs`?

File: test/integration_test.go [hunk 3]
```diff
@@ -807,6 +808,13 @@ func TestExtraArgumentsWithEnv(t *testing.T) {
 	assert.Contains(t, out.String(), "Hello, World!")
 }
 
+func TestExtraArgumentsWithEnvVarBlock(t *testing.T) {
+	out := new(bytes.Buffer)
+	runTerragruntRedirectOutput(t, fmt.Sprintf("terragrunt apply --terragrunt-non-interactive --terragrunt-working-dir %s", TEST_FIXTURE_ENV_VARS_BLOCK_PATH), out, os.Stderr)
+	t.Log(out.String())
+	assert.Contains(t, out.String(), "I'm set in extra_arguments env_vars")
```
Comment (brikis98) [inline test/integration_test.go]: Nice 👍

File: cli/args.go (hunk 2 above)
Comment (geriBatai) [reply cli/args.go]: OK, got it :)

**Comment by brikis98** [comment]
I'll merge now and let the tests run. Once they pass, I'll issue a new release, and paste the link here.

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terragrunt/releases/tag/v0.16.8

**Pull request merged**